src/helpers/update-tasks.ts

```diff
@@ -0,0 +1,172 @@
+
+  if (activity?.length) {
+    const lastCheck = DateTime.fromISO(issue.last_check);
+    const timeDiff = now.diff(lastCheck);
+    const newDeadline = deadline.plus(timeDiff);
+    logger.info(
+      `Activity found on ${issue.url}, will move the deadline forward from ${deadline.toLocaleString(DateTime.DATETIME_MED)} to ${newDeadline.toLocaleString(DateTime.DATETIME_MED)}`
+    );
+    await supabase.repositories.upsert({ url: issue.url, deadline: newDeadline.toJSDate(), lastCheck: now.toJSDate() });
+  } else {
+    if (now >= deadlineWithThreshold) {
+      await unassignUserFromIssue(context, issue);
+    } else if (now >= reminderWithThreshold) {
+      await remindAssigneesForIssue(context, issue);
+    } else {
+      logger.info(
+        `Nothing to do for ${issue.url}, still within due-time (now: ${now.toLocaleString(DateTime.DATETIME_MED)}, reminder ${reminderWithThreshold.toLocaleString(DateTime.DATETIME_MED)}, deadline: ${deadlineWithThreshold.toLocaleString(DateTime.DATETIME_MED)})`
+      );
+    }
+  }
+}
+
+export async function updateTasks(context: Context) {
+  const {
+    adapters: { supabase },
+    logger,
+  } = context;
+  const watchedRepoList = await supabase.repositories.get();
+
+  if (!watchedRepoList?.length) {
+    logger.info("No watched repos have been found, no work to do.");
+    return false;
+  }
+  for (const watchedIssue of watchedRepoList) {
+    await updateReminders(context, watchedIssue);
+  }
+  return true;
+}
+
+async function getAssigneesActivityForIssue({ octokit, payload }: Context, issue: Database["public"]["Tables"]["repositories"]["Row"]) {
+  const { repo, owner, issue_number } = parseGitHubUrl(issue.url);
+  return octokit.paginate(
+    octokit.rest.issues.listEvents,
+    {
+      owner,
+      repo,
+      issue_number,
+      per_page: 100,
+    },
+    (res) =>
+      res.data.filter(
+        (o) =>
+          payload.issue?.assignees?.find((assignee) => assignee?.login === o.actor.login) &&
+          DateTime.fromISO(o.created_at) >= DateTime.fromISO(issue.last_check)
+      )
+  );
+}
+
+async function remindAssignees(context: Context, issue: Database["public"]["Tables"]["repositories"]["Row"]) {
+  const { octokit, logger } = context;
+  const githubIssue = await getGithubIssue(context, issue);
+  const { repo, owner, issue_number } = parseGitHubUrl(issue.url);
+
+  if (!githubIssue?.assignees?.length) {
+    logger.warn(`Missing Assignees from ${issue.url}`);
+    return false;
+  }
+  const logins = githubIssue.assignees
+    .map((o) => o?.login)
+    .filter((o) => !!o)
+    .join(", @");
+  await octokit.rest.issues.createComment({
+    owner,
+    repo,
+    issue_number,
+    body: `@${logins}, this task has been idle for a while. Please provide an update.`,
+  });
+  return true;
+}
+
+async function removeIdleAssignees(context: Context, issue: Database["public"]["Tables"]["repositories"]["Row"]) {
```

Comment:
```suggestion
async function removeAllAssignees(context: Context, issue: Database["public"]["Tables"]["repositories"]["Row"]) {
```
I think this better reflects what the function does